internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -2446,56 +2447,228 @@ func TestComputeDesiredMachine(t *testing.T) {
+					},
+					MachineNamingStrategy: &clusterv1.MachineNamingStrategy{
+						Template: "{{ .machineSet.name }}" + namingTemplateKey,
+					},
+					Template: machineTemplateSpec,
+				},
+			},
+			existingMachine: nil,
+			wantMachine:     nil,
+		},
+		{
+			name: "should not return error when creating a new Machine when the generated name exceeds 63",
+			ms: &clusterv1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: metav1.NamespaceDefault,
+					Name:      msName,
+					Labels: map[string]string{
+						clusterv1.MachineDeploymentNameLabel: mdName,
+					},
+				},
+				Spec: clusterv1.MachineSetSpec{
+					ClusterName:     testClusterName,
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					MachineNamingStrategy: &clusterv1.MachineNamingStrategy{
+						Template: "{{ .random }}" + fmt.Sprintf("%059d", 0),
+					},
+					Template: machineTemplateSpec,
+				},
+			},
+			existingMachine: nil,
+			wantMachine:     expectedNewMachine,
+			wantName: []gomegatypes.GomegaMatcher{
+				ContainSubstring(fmt.Sprintf("%053d", 0)),
+				Not(HaveSuffix("00000")),
+			},
+		},
+		{
+			name: "should return error when creating a new Machine",
+			ms: &clusterv1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: metav1.NamespaceDefault,
+					Name:      msName,
+					Labels: map[string]string{
+						clusterv1.MachineDeploymentNameLabel: mdName,
+					},
+				},
+				Spec: clusterv1.MachineSetSpec{
+					ClusterName:     testClusterName,
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					MachineNamingStrategy: &clusterv1.MachineNamingStrategy{
+						Template: "some-hardcoded-name-{{ .doesnotexistindata }}-{{ .random }}", // invalid template
+					},
+					Template: machineTemplateSpec,
+				},
+			},
+			existingMachine: nil,
+			wantMachine:     nil,
+		},
 		{
-			name:            "creating a new Machine",
+			name: "should return the correct Machine object when creating a new Machine with default templated name",
+			ms: &clusterv1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: metav1.NamespaceDefault,
+					Name:      msName,
+					Labels: map[string]string{
+						clusterv1.MachineDeploymentNameLabel: mdName,
+					},
+				},
+				Spec: clusterv1.MachineSetSpec{
+					ClusterName:     testClusterName,
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					Template: machineTemplateSpec,
+				},
+			},
 			existingMachine: nil,
-			want:            expectedNewMachine,
+			wantMachine:     expectedNewMachine,
+			wantName: []gomegatypes.GomegaMatcher{
+				HavePrefix(msName),
+				Not(HaveSuffix("00000")),
```

Comment:
Is this meant to check if a random number has been added?
If yes, I'm not sure if it works (it wiil pass also if the random number is not added)